gwsumm/plot/segments.py

```diff
@@ -1376,6 +1384,106 @@ def draw(self, outputfile=None):
 register_plot(SegmentBarPlot)
 
 
+class NetworkDutyBarPlot(SegmentBarPlot):
+
+    """Special case of the `SegmentPiePlot` for network duty factors.
+    """
+    type = 'network-duty-segment-bar'
+    NETWORK_NAME = {
+        0: 'no',
+        1: 'single',
+        2: 'double',
+        3: 'triple',
+        4: 'quadruple',
+        5: 'quintuple',
+        6: 'sextuple',
+    }
+    NETWORK_COLOR = GW_OBSERVATORY_COLORS.copy()
+    NETWORK_COLOR.update({
+        'H1L1V1': '#33ff55',
+        'H1L1': '#000000',
+        'L1V1': '#ff33dd',
+        'H1V1': '#ffbb33',
+    })
+    defaults = SegmentBarPlot.defaults.copy()
+    # remove SegmentBarPlot default colors as they overwrite
+    # the ones defined later.
+    defaults.pop('color')
+    defaults.pop('edgecolor')
+    defaults.update({
+        'title': 'Network duty factord sad adas ',
+        'ylabel': 'Duty factor [%]',
+    })
+
+    def draw(self):
+        # get segments
+        if self.state and not self.all_data:
+            valid = self.state.active
+        else:
+            valid = SegmentList([self.span])
+
+        # construct compound flags for each network size
+        flags = dict((f[:2], f) for f in self.flags)
+        # construct all possible network combinations
+        networks = {}
+        for size in range(1, len(flags) + 1):
+            ifocombs = combinations(sorted(set(flags)), size)
+            for ifocomb in ifocombs:
+                key = "".join(ifocomb)
+                networks[key] = (size, ifocomb)
+
+        networkflags = []
+        colors = []
+        labels = []
+        for network, values in networks.items():
+            i = values[0]
+            ifoset = values[1]
+            name = self.NETWORK_NAME[i]
+            if i == 1:
+                # this avoid having a redundant X1:single
+                # in the Segment information table
+                flag = flags[network]
+
+            else:
+                flag = f'{network}:{name}'
+
+                networksegs = DataQualityFlag(flag, known=valid)
+                if not ifoset:
+                    compound = f"!{'!'.join(list(flags.values()))}"
+                else:
+                    compound = '&'.join(flags[ifo] for ifo in ifoset)
+
+                segs = get_segments(compound, validity=valid, query=False,
+                                    padding=self.padding).coalesce()
+                networksegs += segs
+                globalv.SEGMENTS[flag] = networksegs.copy()
+            combined_flag = flag.split(':')[0]
+            if flag.startswith(tuple(networks.keys())):
+                networkflags.append(flag)
+
+            labels.append(combined_flag)
+            if self.NETWORK_COLOR.get(combined_flag) is not None:
+                colors.append(self.NETWORK_COLOR.get(combined_flag))
+            else:
+                # select a random color when it is not defined
+                colors.append(numpy.random.rand(3,))
```

Comment:
Is this behaviour we'd like to have? I'm worried about being able to reproduce the plots the same way each time

Reply:
@iaraota Test!

Reply:
Test2

Reply:
@eagoetz I have changed it to use the [standard matplotlib colors](https://matplotlib.org/stable/gallery/color/named_colors.html#tableau-palette). I also removed the combined detector I had added, because it is only defined in the configuration files. I think that if we want to make them standards they should be added to `gwpy.plot.colors` and added to the other classes. 

Here are the updated examples:

Colors defined in the configuration file: https://ldas-jobs.ligo.caltech.edu/~iara.ota/summary/gps/1384873218-1389456018/segments/

Colors not defines (only single detector colors are set): https://ldas-jobs.ligo.caltech.edu/~iara.ota/summary/gps/1384873218-1389456019/segments/